manifests/http_port.pp

```diff
@@ -1,24 +1,48 @@
 define squid::http_port (
-  Variant[Pattern[/\d+/], Integer]
-          $port    = $title,
-  Boolean $ssl     = false,
-  String  $options = '',
-  String  $order   = '05',
+  Optional[Stdlib::Port] $port    = undef,
+  Optional[String[1]]    $host    = undef,
+  Boolean                $ssl     = false,
+  String                 $options = '',
+  String                 $order   = '05',
 ) {
+  $_title = String($title)
+
+  if $port == undef {
+    if $_title =~ /^(?:.+:)?(\d+)$/ {
+      $_port = Integer($1)
```

Comment:
Ideally you'd also check if it's a valid port now so users don't pass in 100000 instead of 10000.

Reply:
Added as a check against `Stdlib::Port`, with a custom error message.